### PR TITLE
[ownership] Enable ownership lowering /after/ the diagnostic passes.

### PR DESCRIPTION
**Pull request by gottesmm**
I also updated the last group of straggling tests.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci asan test

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
Passed macOS testing. Lets see what happens with iphonesimulator. I am looking into the swiftpm and linux failure.

**Comment by gottesmm** [comment]
I have a solution for the swiftpm issue.

**Comment by gottesmm** [comment]
Hit some lldb issues. LLDB fix is here: https://github.com/apple/swift/pull/28042.

Also the macOS tests passed the full test.

**Comment by gottesmm** [comment]
Landed the lldb change (just needed to lower ownership). Curious if I get significantly further this time on the macOS test (ignoring the swiftpm failure that I am going to hit).

**Comment by gottesmm** [comment]
@swift-ci test os x platform

**Comment by gottesmm** [comment]
OS X failure was this:

15:55:44 <unknown>:0: note: Cannot read '/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/buildbot_incremental/swift-macosx-x86_64/./lib/swift/iphoneos/layouts-arm64.yaml'?

I think that might mean a bad node? Going to try again.

**Comment by gottesmm** [comment]
@swift-ci test os x platform

**Comment by gottesmm** [comment]
Looked at the ASAN failure. Was a source kit error. Definitely not mine:

15:04:12 --
15:04:12 Exit Code: 1
15:04:12 
15:04:12 Command Output (stderr):
15:04:12 --
15:04:12 
/Users/buildnode/jenkins/workspace/swift-PR-osx-ASAN-test/swift/test/SourceKit/CursorInfo/invalid_offset.swift:15:11: error: CHECK: expected string not found in input
15:04:12 // CHECK: (Request Failed): Unable to resolve the start of the token
15:04:12           ^
15:04:12 <stdin>:1:1: note: scanning from here
15:04:12 {
15:04:12 ^
15:04:12 <stdin>:34:30: note: possible intended match here
15:04:12 <empty cursor info; internal diagnostic: "Unable to resolve cursor info.">
15:04:12                              ^
15:04:12 
15:04:12 --
15:04:12 
15:04:12 ******.

Everything else passed though.

**Comment by gottesmm** [comment]
Ran the lldb test again with my failure. Failed again, but only in the ref count test:

18:04:16     lldb-Suite :: lang/swift/swift_reference_counting/TestSwiftReferenceCount.py

The failure is since lldb is testing exact reference counts and after my changes the ref count is 4 instead of 3.

**Comment by gottesmm** [comment]
#28208 should fix the swiftpm issue. I discovered some similar fixes I want to do on the load [copy] side, but the load_borrow fix there should take care of that.

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
I think I am going to hit an lldb test failure, but the swiftpm issue is fixed.

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

**Comment by gottesmm** [comment]
Linux hit the option set failure. I think that just needs a small test update.

**Comment by gottesmm** [comment]
I am expecting an lldb test failure as well.

**Comment by gottesmm** [comment]
But on the macOS side. I need to provide a separate PR to test with this that updates that lldb test.

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
I am fixing another bug right now, but I am going to loop back on this in a bit with the lldb patch/linux fix.

**Comment by gottesmm** [comment]
OS X failed on the lldb test as expected, but all of the normal tests passed.

**Comment by gottesmm** [comment]
I sat down with Davide and un-hardcoded the check for the specific ref count of the object in the test and instead just made sure we can properly get output. The fix is here: apple/llvm-project#294. I am able to land it before landing this so I don't need to do cross repo testing.

**Comment by gottesmm** [comment]
apple/llvm-project#294

@swift-ci test

**Comment by gottesmm** [comment]
Trying the OS X test but this time with my lldb patch (hasn't landed yet, but I want to get a clean OS X run).

**Comment by gottesmm** [comment]
Linux test failure is expected since I haven't fixed that small test that is failing. I think the test is just too specific (its a check-next line failing).

**Comment by gottesmm** [comment]
From the debug source compat suite:

17:24:44 Action Summary:
17:24:44      Passed: 409
17:24:44      Failed: 0
17:24:44     XFailed: 5
17:24:44     UPassed: 1
17:24:44       Total: 415
17:24:44 ====================

So it passed. I think the UPass is Bow that was just XFAILed recently.

**Comment by gottesmm** [comment]
MacOS full tests passed

**Comment by gottesmm** [comment]
@swift-ci asan test

**Comment by gottesmm** [comment]
Release source compat:

21:56:53 ========================================
21:56:53 Action Summary:
21:56:53      Passed: 406
21:56:53      Failed: 0
21:56:53     XFailed: 8
21:56:53     UPassed: 1
21:56:53       Total: 415
21:56:53 ========================================

I think this is another case where Bow needs to be unXFAILed.

**Comment by gottesmm** [comment]
So I think this means once I fix that one linux test, we are ready to roll in terms of the PR.

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
Windows failed in the same way as Linux on that optionset test having different output. Going to look at that today after I finish the load [copy] version of the load_borrow change.

**Comment by gottesmm** [comment]
Figured out what was up with the optionset test. Fixed it up.

**Comment by gottesmm** [comment]
I just gotta see what happened with the non-determinism issue with the load [copy] patch.

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
I am going to run some tests overnight that test out this with: #28314. That way once it lands I can merge (*cross fingers*)

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci asan test

**Comment by gottesmm** [comment]
@swift-ci asan test

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
@swift-ci asan test

**Comment by gottesmm** [comment]
Got full greens using full tests.. Going to grab a little zzzs, then see if I can quickly fix the os log test and then smoke and merge.

**Comment by gottesmm** [comment]
I talked with Ravi. I am going to commit that test disabled so I can land this and I am going to post a patch with the post-world fix that Ravi can review.

**Comment by gottesmm** [comment]
Asan, source compat, full Mac/Linux testing, and windows too! I’m going to clean up the git history a little and then smoke test and merge.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by drodriguez** [comment]
Good morning/evening/night,

I was looking into the failures that appeared in https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android/4509/ (Android ARMv7) after the introduction of this patch. It seems that the change `bool StripOwnershipAfterSerialization = false/true;` makes the test `test/Frontend/skip-non-inlinable-function-bodies.swift` introduced in #20420 fail. It seems, but I am not sure, that the output after this change is introduced avoids the setter to be inlined. Does that makes sense?

I can XFAIL this test, but I will not mind a ELI5 of what's going on to understand the problem better, and maybe find an actual solution.

The two outputs (before and after the patch) are in https://gist.github.com/drodriguez/658c0e0818e1d89c03633150d9d858b1

/cc @harlanhaskins

**Comment by harlanhaskins** [comment]
Text of the failure:

```
Command Output (stderr):
--
/home/ubuntu/jenkins/workspace/oss-swift-RA-linux-ubuntu-16.04-android/swift/test/Frontend/skip-non-inlinable-function-bodies.swift:160:49: error: CHECK: expected string not found in input
 _blackHole("@inlinable deinit body") // CHECK: @inlinable deinit body
                                                ^
<stdin>:1790:51: note: scanning from here
 %36 = string_literal utf8 "@inlinable setter body" // user: %39
                                                  ^
<stdin>:1831:39: note: possible intended match here
} // end sil function '$s4main6StructV15inlinableSetterSivsTf4dn_n'
                                      ^

--

********************
```

It looks like the deinit wasn't emitted, maybe that's a side effect of ownership visiting slightly differently?

**Comment by harlanhaskins** [comment]
Anything marked `@inlinable` needs to be SILGen'd.

**Comment by drodriguez** [comment]
The deinit was emitted, but it is emitted before the setter, but because the setter is checked before, the deinit is not found _below_ the setter, and the test fails. But I imagine is something like you say, that the ownership being enabled or disabled changes the visiting order slightly… and only in ARMv7? (32 bits vs 64 bits?)

**Comment by harlanhaskins** [comment]
Ah, I see. That's...not very great. I should probably be passing -emit-sorted-sil...

**Comment by drodriguez** [comment]
Let me check if that solves the problem. Thanks for the clue!
